artemis-ra/src/main/java/org/apache/activemq/artemis/ra/ActiveMQRAConnectionManager.java

```diff
@@ -39,7 +39,7 @@ public ActiveMQRAConnectionManager() {
       logger.trace("constructor()");
    }
 
-   ConcurrentHashSet<ManagedConnection> connections = new ConcurrentHashSet<>();
+   transient ConcurrentHashSet<ManagedConnection> connections = new ConcurrentHashSet<>();
```

Comment:
My recollection of Serialization code (which is pretty foggy) is that when you do this the value of the connection variable will be null after readObject is called regardless of the initialization here and you would need to add an override of readObject to initialize the value of connections:

```
    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
        in.defaultReadObject();

        connections = new ConcurrentHashSet<>();
    }
```

Same holds true for some of the other new transient collections added below

Reply:
I applied fixes in all the relevant locations.